### DISCUSSION TITLE
Alpha channel is lost after resizing and cropping if the original image has entirely black alpha channel

### DISCUSSION THREAD

**Discussion by MishaPhysics** [ImageMagick/ImageMagick#7324] [Help] [answered]
### ImageMagick version

7.1.1-14

### Operating system, version and so on

Windows 10

### Description

I have the original image "file.bmp" of 16384x16384 with an alpha channel which is entirely black (in Photoshop). After both these commands:

```
magick mogrify -resize 8192x8192! *.bmp
magick mogrify -crop 512x512 *.bmp
```

I get pictures without the alpha channel, so it is lost. It is a big problem for me. Hope there are some options (ways) to "fix" it. (?)

But when I use the original image with an alpha channel which is not entirely black, then everything is good, and the files after resizing and cropping have the alpha channel even if some files have entirely black alpha channel.

(If it is important, then I convert my cropped *.bmp files to *.dds format using:

`magick mogrify -format dds *.bmp`

But the problem seems to be with the resizing and cropping commands applying to the input image with the entirely black alpha channel.)

**Comment by urban-warrior** [comment]
Unfortunately, we cannot reproduce your issue with the latest ImageMagick release, 7.1.1-32.   Try 7.1.1-32 and if it still fails, first try `magick identify -verbose *.bmp` to confirm the alpha channel is indeed missing.  If so, post a link to your image here.  We will need to download it and attempt to reproduce the issue.

**Reply by MishaPhysics** [reply]
Thanks. I have installed "ImageMagick-7.1.1-32-Q16-HDRI-x64-dll.exe" over the old version. Maybe I had to unistall the old version before the new installation, so I am not sure the new version was installed correctly because when I am trying to open "ImageMagick Display" or when I am trying to open a picture with ImageMagick I have got the "Entry Point Not Found Error".

Commands in terminal work, but the alpha channel is still missing.

I have tried `magick identify -verbose *.bmp`. But I am not sure I understand that information.

Here is my picture (I reduced its size to 1024x1024). It has an alpha channel (I see it in Photoshop):
[1024.zip](https://github.com/ImageMagick/ImageMagick/files/15369496/1024.zip)
After running the command:
`magick mogrify -resize 512x512! *.bmp`
or
`magick mogrify -crop 512x512 *.bmp`
the output pictures have no alpha channel.

**Reply by urban-warrior** [reply]
Thank you for reporting the issue. We have successfully reproduced it and are actively working on a patch to resolve it. You can expect this patch to be merged into the [main GIT branch](https://github.com/ImageMagick/ImageMagick), later today. As part of our commitment to quality, this fix will also be included in the upcoming [beta releases of ImageMagick](https://imagemagick.org/archive/beta/) by tomorrow. Your patience and feedback are greatly appreciated.

**Reply by MishaPhysics** [reply]
Thanks, I will wait.
Also, I have noticed the following, maybe it may help. I have added one white pixel to my entirely black alpha channel in the top left corner of the image, and it works, namely the alpha channel is saved after resizing and cropping.

Sorry, I'm afraid I haven't understood where I could find that updated (in the future) ImageMagick you mentioned above and how to update it when it will be available, namely what file I will need.

**Reply by urban-warrior** [reply]
The patch will be available in ImageMagick 7.1.1-33 scheduled for release next weekend.

**Reply by MishaPhysics** [reply]
Excuse me, I just wanted to ask if that patch with a fix for an alpha channel is already available for trying, or it will be later? I've seen a 7.1.1-33 versions in beta releases, but I don't know what file I need to install for my Windows 10 (64-bit).

**Reply by fmw42** [reply]
You can check here for the binary releases https://imagemagick.org/script/download.php#windows

**Reply by MishaPhysics** [reply]
There's only the 7.1.1-32 versions right now. Do I understand correctly, that there will be the 7.1.1-33 one later? Thanks for your answer.

**Reply by fmw42** [reply, marked as answer]
Yes, that is correct.  The next one often comes out a few weeks after the current one. It just depends upon a number of factors including urgency, what has been do in the way of bug fixes and the developers time availability.

**Reply by MishaPhysics** [reply]
I've just tried the 7.1.1-33 binary release for Windows after deleting my old ImageMagick version. The alpha channel issue is solved now. But there's no the ImageMagick viewer after installation, so I can't open a picture with the ImageMagick. It is expected? Also, I've got the error just after installation (I checked all the boxes before installation):

![error](https://github.com/ImageMagick/ImageMagick/assets/104359915/5fa39220-8961-440d-bcdb-5315329d68de)
